### ISSUE TITLE
`mqtt/superuser` POST called two times in `emqtt_auth_http`

### ISSUE THREAD

**Issue by drasko** [emqtt/emqttd#696]
Hello,
I am observing following behaviour:

1. When trying:
```
mosquitto_pub -h localhost -t mainflux/66be841c-8e9e-4620-b0b6-7c4b5ed823eb -m "123" -u blah -P TEST
```

I get:
```
127.0.0.1 - - [13/Sep/2016 20:51:21] "POST /mqtt/superuser HTTP/1.1" 405 -
127.0.0.1 - - [13/Sep/2016 20:51:21] "POST /mqtt/auth HTTP/1.1" 200 -
127.0.0.1 - - [13/Sep/2016 20:51:21] "POST /mqtt/superuser HTTP/1.1" 405 -
127.0.0.1 - - [13/Sep/2016 20:51:21] "POST /mqtt/acl HTTP/1.1" 200 -
```

I have put the function that handles `/mqtt/superuser` path to fail on purpose in my auth server, so I can observe what is happening with user auth.

We can see that POST to `/mqtt/superuser` is doubled.

Also,
when I change my HTTP auth server to return `200` on `/mqtt/superuser` POST, I am seeing:
```
127.0.0.1 - - [13/Sep/2016 20:56:14] "POST /mqtt/superuser HTTP/1.1" 200 -
127.0.0.1 - - [13/Sep/2016 20:56:14] "POST /mqtt/superuser HTTP/1.1" 200 -
```

So - eMQTT does two POST req towards my auth server.

Why is this happening - is not only one POST request for superuser enough?

**Comment by huangdan** [comment]
@drasko when doing the ACL and Auth check,will make the client is superUsers judgment

**Comment by emqplus** [comment]
@drasko There are two '/mqtt/superuser' requests in the current design. One comes from Authentication when a client connected to the broker, and the other from Authorization(ACL)when publishing a packet.

**Comment by drasko** [comment]
@emqplus I see, but is not this other redundant?

If I am listening on the `/mqtt/superuser` - how will I know which is for Auth and which is for ACL?

Moreover - do we even need one for ACL in the case of superuser, as he has unlimited access?

**Comment by emqplus** [comment]
@drasko, I improved the design of Auth/ACL in 2.0-rc.1. Only one `mqtt/superuser` call for superuser now.

**Comment by drasko** [comment]
@emqplus thanks, we will test it.

**Comment by drasko** [comment]
Not fixed in `emqttd-debian64-v2.0-beta.3-20160918.zip`. Please re-open the bug.
